google/_async_resumable_media/requests/_request_helpers.py

```diff
@@ -0,0 +1,155 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Shared utilities used by both downloads and uploads.
+
+This utilities are explicitly catered to ``requests``-like transports.
+"""
+
+
+import functools
+
+from google._async_resumable_media import _helpers
+from google.resumable_media import common
+
+import google.auth.transport._aiohttp_requests as aiohttp_requests
```

Comment:
This, which is used to allow for a combined response, has a hard dependency on google-auth. We don't currently have one for the package.

Reply:
Should `google-auth` be re-added as an explicit dependency?

Reply:
I think it is safe to do so. It is unlikely this library is used without google-auth or something else that requires it (api-core for instance).

@frankyn @tswast any opinions?

Reply:
https://github.com/googleapis/google-resumable-media-python/issues/180 I made thinking we might need this.

I sort of think refactoring to avoid the dependency may be futile since folks will have the dependency anyway?

Reply:
@crwilcox I agree - if most folks are going to have to install `google-auth` I'd prefer to have it in the requirements.